llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -31040,17 +31044,63 @@ static SDValue LowerBITREVERSE_XOP(SDValue Op, SelectionDAG &DAG) {
   return DAG.getBitcast(VT, Res);
 }
 
+static auto createBSWAPShuffleMask(EVT VT) {
```

Comment:
Style: We avoid to use `auto` in function definition. And it's better to pass a `SmallVectorImpl<int> &ShuffleMask` to avoid copy.

Reply:
I have updated the function definition like you suggested. However, I think, because of RVO (Return Value Opitmization/Copy elision), the copy would not have been made.